### ISSUE TITLE
Warn if first word of a `for` loop is an unquoted known command name

### ISSUE THREAD

**Issue by koalaman** [koalaman/shellcheck#1230]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

Sometimes people try to loop over command output like this:

```sh
#!/bin/sh
for f in seq 1 10
do true; done
```

Similarly, `for i in ls foo`, `for f in cat file.txt`.

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

To iterate over command output, use $(..) or a while read loop (or quote command name to silence).

This may need special cases for `true/false` (ignore) and `ls` (use globs). It may also be useful to try to ignore typical dependency checks like `for i in awk sed grep; do ..` by checking that the second word is *not* a command.
